src/applications/vaos/new-appointment/redux/actions.js

```diff
@@ -464,14 +464,18 @@ export function openFacilityPageV2(page, uiSchema, schema) {
 }
 
 export function updateCCProviderSortMethod(sortMethod) {
-  return async (dispatch, _getState) => {
+  return async (dispatch, getState) => {
     let location = null;
+    const { currentLocation } = getNewAppointment(getState());
     const action = {
       type: FORM_PAGE_CC_FACILITY_SORT_METHOD_UPDATED,
       sortMethod,
     };
 
-    if (sortMethod === FACILITY_SORT_METHODS.distanceFromCurrentLocation) {
+    if (
+      sortMethod === FACILITY_SORT_METHODS.distanceFromCurrentLocation &&
+      Object.keys(currentLocation).length === 0
```

Comment:
Skip fetching the location again if we already have it, since we're not going to refetch the list even if the location changes.